lib/ecmascript.ts

```diff
@@ -1294,798 +1484,489 @@ export const ES = ObjectAssign({}, ES2020, {
+  RejectYearMonthRange(isoYear, isoMonth);
+
+  CreateSlots(result);
+  SetSlot(result, ISO_YEAR, isoYear);
+  SetSlot(result, ISO_MONTH, isoMonth);
+  SetSlot(result, ISO_DAY, referenceISODay);
+  SetSlot(result, CALENDAR, calendar);
+  SetSlot(result, YEAR_MONTH_BRAND, true);
+
+  if (DEBUG) {
+    Object.defineProperty(result, '_repr_', {
+      value: `${result[Symbol.toStringTag]} <${TemporalYearMonthToString(result)}>`,
+      writable: false,
+      enumerable: false,
+      configurable: false
+    });
+  }
+}
+
+export function CreateTemporalYearMonth(isoYear, isoMonth, calendar = GetISO8601Calendar(), referenceISODay = 1) {
+  const TemporalPlainYearMonth = GetIntrinsic('%Temporal.PlainYearMonth%');
+  const result = ObjectCreate(TemporalPlainYearMonth.prototype);
+  CreateTemporalYearMonthSlots(result, isoYear, isoMonth, calendar, referenceISODay);
+  return result;
+}
+
+export function CreateTemporalZonedDateTimeSlots(result, epochNanoseconds, timeZone, calendar) {
+  ValidateEpochNanoseconds(epochNanoseconds);
+
+  CreateSlots(result);
+  SetSlot(result, EPOCHNANOSECONDS, epochNanoseconds);
+  SetSlot(result, TIME_ZONE, timeZone);
+  SetSlot(result, CALENDAR, calendar);
+
+  const TemporalInstant = GetIntrinsic('%Temporal.Instant%');
+  const instant = new TemporalInstant(GetSlot(result, EPOCHNANOSECONDS));
+  SetSlot(result, INSTANT, instant);
+
+  if (DEBUG) {
+    Object.defineProperty(result, '_repr_', {
+      value: `${result[Symbol.toStringTag]} <${TemporalZonedDateTimeToString(result, 'auto')}>`,
+      writable: false,
+      enumerable: false,
+      configurable: false
+    });
+  }
+}
+
+export function CreateTemporalZonedDateTime(epochNanoseconds, timeZone, calendar = GetISO8601Calendar()) {
+  const TemporalZonedDateTime = GetIntrinsic('%Temporal.ZonedDateTime%');
+  const result = ObjectCreate(TemporalZonedDateTime.prototype);
+  CreateTemporalZonedDateTimeSlots(result, epochNanoseconds, timeZone, calendar);
+  return result;
+}
+
+export function GetISO8601Calendar() {
+  const TemporalCalendar = GetIntrinsic('%Temporal.Calendar%');
+  return new TemporalCalendar('iso8601');
+}
+
+export function CalendarFields(calendar, fieldNames) {
+  if (calendar.fields) {
+    fieldNames = calendar.fields(fieldNames);
+  }
+  const result = [];
+  for (const name of fieldNames) {
+    if (Type(name) !== 'String') throw new TypeError('bad return from calendar.fields()');
+    ArrayPrototypePush.call(result, name);
+  }
+  return result;
+}
+
+export function CalendarMergeFields(calendar, fields, additionalFields) {
+  const calMergeFields = calendar.mergeFields;
+  if (!calMergeFields) {
+    return { ...fields, ...additionalFields };
+  }
+  const result = Reflect.apply(calMergeFields, calendar, [fields, additionalFields]);
+  if (Type(result) !== 'Object') throw new TypeError('bad return from calendar.mergeFields()');
+  return result;
+}
+
+export function CalendarDateAdd(calendar, date, duration, options, dateAdd?: any) {
```

Comment:
possible to replace `any` with a real type?

Reply:
Will do in a follow-up - this is just to get tsc to allow calling this without that particular parameter.